test/e2e/apimachinery/garbage_collector.go

```diff
@@ -67,7 +67,7 @@ func estimateMaximumPods(c clientset.Interface, min, max int32) int32 {
 		availablePods += 10
 	}
 	//avoid creating exactly max pods
-	availablePods *= 8 / 10
+	availablePods = int32(float32(availablePods) * 8.0 / 10.0)
```

Comment:
does this function have unit tests, because this is a change in behavior?
the original code has a problem; the rvalue `8 / 10` is always zero as per the lvalue type of int32.

```
	var availablePods int32
	availablePods = 15
	availablePods *= 8 / 10
	fmt.Println(availablePods) // 0
	availablePods = 15
	availablePods = int32(float32(availablePods) * 8.0 / 10.0)	
	fmt.Println(availablePods) // 12
```

also the correct order of operations in your change should not be `*` then `/`.
(most) compilers evaluate the rvalue before the operator+assignment with the lvalue.

it should be `availablePods = int32(float32(availablePods) * (8.0 / 10.0))`.
```
	var availablePods float32
	availablePods = 15
	availablePods *= 10.0
	availablePods /= 8.0
	fmt.Println(availablePods) // 18.75
	availablePods = 15
	availablePods *= 8.0 / 10.0
	fmt.Println(availablePods) // 12
```

but since the original code is buggy it's unclear whether truncating the float is the desired result.
possibly, yes.

Reply:
I don't see any unit tests for this function; it itself is part of a test. And yes, it was always being set to zero before (and then would be set to `min` at line 76), which I too assumed was a bug and I think the desired behavior was to multiply it by .8 but we could check with the original author if we want to be sure.

Thanks for the catch on the order of operations - I will update that!

Reply:
it can be changed to ` * 0.8` too.
original commit is here:
https://github.com/kubernetes/kubernetes/commit/91cc7b94c8f4af77c5fd8814761419de2fb63ac8

Reply:
@smarterclayton the commit in question is from 4 years ago haha but any chance you could give a confirmation as to whether it's ok here to change the behavior to actually multiply by 0.8 instead of setting to 0? Maybe silly to ask but just want to be sure :)